### PR TITLE
MC-12984 : Added API docs

### PR DESCRIPTION
**Pull request by fyrnaga**
#### Changes made

- Added documentation on stackpath plugin for list, get and restart instances
- Fixed small error id workload doc

#### Related PRs
- PR # https://github.com/cloudops/cloudmc-api-docs/pull/264
- PR # https://github.com/cloudops/cloudmc-stackpath-plugin/pull/10

### CODE REVIEW

File: source/includes/stackpath/_instances.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      "status": "RUNNING"
+    }
+  ],
+  "metadata": {
+    "recordCount": 1
+  }
+}
+```
+
+<code>GET /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/instance?workload=:worlloadId</code>
```
Comment (y0geshdev) [inline source/includes/stackpath/_instances.md]: a small type in `?workload=:worlloadId`

File: source/includes/stackpath/_instances.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+Retrieve a list of all workload instance in a given [environment](#administration-environments) for a given workload id.
+
+Query Params | &nbsp;
+---- | -----------
+`workloadId`<br/>*string* | The workload load id to get the instances for. It is mandatory.
```
Comment (y0geshdev) [inline source/includes/stackpath/_instances.md]:
```suggestion
`workloadId`<br/>*string* | The workload id to get the instances for. It is mandatory.
```

File: source/includes/stackpath/_instances.md (hunk 1 above)
Comment (fyrnaga) [reply source/includes/stackpath/_instances.md]: ef5640da7d2e6dc2974054f6465e93ead7fdd42f

File: source/includes/stackpath/_instances.md (hunk 2 above)
Comment (fyrnaga) [reply source/includes/stackpath/_instances.md]: ef5640da7d2e6dc2974054f6465e93ead7fdd42f

File: source/includes/stackpath/_instances.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+Query Params | &nbsp;
+---- | -----------
+`workloadId`<br/>*string* | The workload ID to get the instances for. It is mandatory.
+
+Attributes | &nbsp;
+------- | -----------
+`id`<br/>*string* | The instance id. It is the combination of the instance workloadId and instance name.
+`name`<br/>*string* | The instance name.
+`status`<br/>*string* | The instance status. Possible values are: RUNNING, STARTING, SCHEDULING.
+`stackId`<br/>*UUID* | The stack ID to which belongs the instance.
```
Comment (y0geshdev) [inline source/includes/stackpath/_instances.md]: Would it make it a bit more clear if we reword it as - `The stack ID to which this instance belongs.`?

File: source/includes/stackpath/_instances.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+`workloadId`<br/>*UUID* | The workload ID to which belongs the instance.
```
Comment (y0geshdev) [inline source/includes/stackpath/_instances.md]: Same as above.

File: source/includes/stackpath/_instances.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+Retrieve an instance in a given [environment](#administration-environments) with the specific id.
+
+Attributes | &nbsp;
+------- | -----------
+`id`<br/>*string* | The instance id. It is the combination of the instance workloadId and instance name.
+`name`<br/>*string* | The instance name.
+`status`<br/>*string* | The instance status. Possible values are: RUNNING, STARTING, SCHEDULING.
+`stackId`<br/>*UUID* | The stack ID to which belongs the instance.
+`workloadId`<br/>*UUID* | The workload ID to which belongs the instance.
```
Comment (y0geshdev) [inline source/includes/stackpath/_instances.md]: Same thought as above.

**Review by y0geshdev — COMMENTED** [review]
Looks good. Just a few minor queries.

File: source/includes/stackpath/_instances.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
+### Instances
```
Comment (patrickspensieri) [inline source/includes/stackpath/_instances.md]: I might be wrong, but I think there's an extra level of nesting here. If true, this should be `## Instances` and the operations should be `### List instances` for example

File: source/includes/stackpath/_instances.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+Retrieve a list of all workload instance in a given [environment](#administration-environments) for a given workload id.
+
+Query Params | &nbsp;
+---- | -----------
+`workloadId`<br/>*string* | The workload ID to get the instances for. It is mandatory.
+
+Attributes | &nbsp;
+------- | -----------
+`id`<br/>*string* | The instance id. It is the combination of the instance workloadId and instance name.
```
Comment (patrickspensieri) [inline source/includes/stackpath/_instances.md]: For consistency, can capitalize ID here. Also, can specify the format with something like `<workloadId>-<instanceName>`. (Also update other tables)

**Review by patrickspensieri — COMMENTED** [review]
Looks good, just a small question regarding the levels of nesting

File: source/includes/stackpath/_instances.md (hunk 6 above)
Comment (fyrnaga) [reply source/includes/stackpath/_instances.md]: yep. you are right. I base myself on the workload. I will change that.

File: source/includes/stackpath/_instances.md (hunk 3 above)
Comment (fyrnaga) [reply source/includes/stackpath/_instances.md]: will change all of them

File: source/includes/stackpath/_instances.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+`name`<br/>*string* | The instance name.
+`status`<br/>*string* | The instance status. Possible values are: RUNNING, STARTING, SCHEDULING.
+`stackId`<br/>*UUID* | The stack ID to which belongs the instance.
+`workloadId`<br/>*UUID* | The workload ID to which belongs the instance.
+`ipAddress`<br/>*string* | The internal IP address associated to the instance.
```
Comment (danielvanacker) [inline source/includes/stackpath/_instances.md]: to --> with (same as below)

File: source/includes/stackpath/_instances.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+Retrieve an instance in a given [environment](#administration-environments) with the specific id.
+
+Attributes | &nbsp;
+------- | -----------
+`id`<br/>*string* | The instance id. It is the combination of the instance workloadId and instance name.
+`name`<br/>*string* | The instance name.
+`status`<br/>*string* | The instance status. Possible values are: RUNNING, STARTING, SCHEDULING.
+`stackId`<br/>*UUID* | The stack ID to which belongs the instance.
+`workloadId`<br/>*UUID* | The workload ID to which belongs the instance.
+`ipAddress`<br/>*string* | The internal IP address associated to the instance.
```
Comment (danielvanacker) [inline source/includes/stackpath/_instances.md]: Same as above to --> with

File: source/includes/stackpath/_instances.md (hunk 3 above)
Comment (fyrnaga) [reply source/includes/stackpath/_instances.md]: b0201713923a94550fc763500a5a5a0a264e51b3

File: source/includes/stackpath/_instances.md (hunk 4 above)
Comment (fyrnaga) [reply source/includes/stackpath/_instances.md]: b0201713923a94550fc763500a5a5a0a264e51b3

File: source/includes/stackpath/_instances.md (hunk 5 above)
Comment (fyrnaga) [reply source/includes/stackpath/_instances.md]: b0201713923a94550fc763500a5a5a0a264e51b3

File: source/includes/stackpath/_instances.md (hunk 6 above)
Comment (fyrnaga) [reply source/includes/stackpath/_instances.md]: b0201713923a94550fc763500a5a5a0a264e51b3

File: source/includes/stackpath/_instances.md (hunk 7 above)
Comment (fyrnaga) [reply source/includes/stackpath/_instances.md]: e27dc4c55ea93c46d14d801c5736954b2954be64

File: source/includes/stackpath/_instances.md (hunk 8 above)
Comment (fyrnaga) [reply source/includes/stackpath/_instances.md]: e27dc4c55ea93c46d14d801c5736954b2954be64

File: source/includes/stackpath/_instances.md (hunk 9 above)
Comment (fyrnaga) [reply source/includes/stackpath/_instances.md]: e27dc4c55ea93c46d14d801c5736954b2954be64

**Review by y0geshdev — APPROVED** [review]
Thanks for the changes. Looks good.

**Review by danielvanacker — APPROVED** [review]

**Review by patrickspensieri — APPROVED** [review]
Nice work!

**Pull request merged**